contracts/transfer/tests/common/prove.rs

```diff
@@ -0,0 +1,155 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_bytes::Serializable;
+use execution_core::{
+    plonk::{Circuit, Prover},
+    transfer::phoenix::{
+        InputNoteInfo, Prove, TxCircuit, TxCircuitVec, NOTES_TREE_DEPTH,
+    },
+};
+
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+
+pub struct CachedProver();
```

Comment:
This looks like the exact same code, right? Maybe we could – not necessary now – avoid duplication by moving this in an higher level of the crates hierarchy in the workspace, so both contracts' test can access to it.

Reply:
The tests for both the stake and transfer contracts share a large amount of code which atm is simply duplicated in both crates. It is one of the tech-debts that I chose not to tackle yet.

And additionally: the `Prove` trait from execution-core will replace the `rusk-prover::Prover` trait. The lib `rusk-prover` will then be a specific implementation of the `Prove` trait and at that point can be used by the contracts in their tests.
This will remove these two files.
See #2105